client/landing/gutenboarding/stores/domain-suggestions/types.ts

```diff
@@ -33,13 +33,15 @@ export interface DomainSuggestionQuery {
 	vertical?: string;
 }
 
+export type DomainName = string;
```

Comment:
In a _structural_ type system (like TypeScript), this has no effect. All the compiler and type system see is `string` (including completion in the editor).

Many other type systems are _nominal_, so if we give a `string` another name, it's unique and no longer a string.

There's a joke about a rose in there 🤔 😄 

This sort of alias is a practice we've been using to communicate with developers. While the type system doesn't care, `DomainName` is intended to indicate something more than just `string`.

We're still figuring out best practices and experience will guide us about when or if to use this sort of technique.

If you think this is helpful or a hindrance, I'd love to know 🙂

Reply:
> In a _structural_ type system (like TypeScript), this has no effect. All the compiler and type system see is `string` (including completion in the editor).
> 
> Many other type systems are _nominal_, so if we give a `string` another name, it's unique and no longer a string.
> 
> There's a joke about a rose in there  
> 
> This sort of alias is a practice we've been using to communicate with developers. While the type system doesn't care, `DomainName` is intended to indicate something more than just `string`.

Yeah, inded. I've found it documented in our [TS docs](https://github.com/Automattic/wp-calypso/blob/9a1aebce0b4bae71bf7ef5079240c6363a9b7101/docs/coding-guidelines/typescript.md#primitive-types), (c) Dennis.

I also personally think this makes sense; you could argue that not every string is a `DomainName`, and that we might add some kind of assertions for them. (I think we actually use some heuristics in other, non-typed, parts of our codebase, in order to infer whether a route fragment is a site slug; that might serve as an analogy here.)

And even if none of the latter applies, I don't mind documenting that an argument is a `DomainName` rather than 'just' a `string` -- even if the latter is just an alias for the former.

> We're still figuring out best practices and experience will guide us about when or if to use this sort of technique.
> 
> If you think this is helpful or a hindrance, I'd love to know

:heavy_plus_sign: :one:

Reply:
It's helpful and it makes sense. Thanks for the detailed explanation!
So when a property of an interface can be described better than just by using a primitive, we should use a type alias for it. I'll keep this in mind. 👍

Reply:
I suspect there's some point where we overuse this and all the types just become nonsense, but we'll feel this out 😉

Reply:
> There's a joke about a rose in there

:rose: === :rose: === :rose: